### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by qchouleur**

### CODE REVIEW

File: MyHome.DataClasses/Expense.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    (Comment == expenseComparing.Comment) &&
+                    (Date == expenseComparing.Date) &&
+                    (Id == expenseComparing.Id) &&
+                    (Method.Equals(expenseComparing.Method)));
+        }
+
+        // TODO - QC - override get hash code as well.
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```
Comment (M-Zuber) [inline MyHome.DataClasses/Expense.cs]: A simple way to do this is also override `ToString`, and then return the `this.ToString().GetHashCode()`

File: MyHome.DataRepositories/GeneralCategoryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MyHome.Persistence;
+
+namespace MyHome.DataRepository
+{
+    // TODO - QC - Create a service that merges the categories
```
Comment (M-Zuber) [inline MyHome.DataRepositories/GeneralCategoryHandler.cs]: In what way?

File: MyHome.Services/CategoryService.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace MyHome.Services
+{
+    public class CategoryService
+    {
+        private readonly Dictionary<int, ICategoryService> _categoryServicesById;
+        
+        public CategoryService(AccountingDataContext context)
+        {
+            var context1 = context;
```
Comment (M-Zuber) [inline MyHome.Services/CategoryService.cs]: Why do we need a temp variable here?

File: MyHome.Services/ExpenseCategoryService.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            _repository.RemoveByName(name);
+        }
+
+        public void Update(int id, string name)
+        {
+            Contract.Requires<ArgumentException>(!string.IsNullOrEmpty(name));
+
+            if (Exists(name))
+            {
+                throw new Exception(string.Format("Expense category '{0}' is already defined", name));
```
Comment (M-Zuber) [inline MyHome.Services/ExpenseCategoryService.cs]: Maybe preferable to return success/failure value that can more easily be consumed by the UI?
Applies to all `*Service`

File: MyHome.Services/MonthService.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            categoryTotals.AddRange(_expenseService.GetAllCategoryTotals(month));
+
+            categoryTotals.Add("Total Income", GetTotalIncomeForMonth(month));
+            foreach (var totalIncomeByCategory in _incomeService.GetTotalIncomeByCategories(month))
+            {
+                if (categoryTotals.ContainsKey(totalIncomeByCategory.Key))
+                {
+                    var placeholder = categoryTotals[totalIncomeByCategory.Key];
+                    categoryTotals.Remove(totalIncomeByCategory.Key);
+#warning Verify that part.. >.<
```
Comment (M-Zuber) [inline MyHome.Services/MonthService.cs]: Meaning?

File: MyHome.UI/Helpers/HelperMethods.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿namespace MyHome.UI.Helpers
+{
+    public static class HelperMethods
+    {
+        public static bool IsDouble(this string input)
+        {
+            double dbToParse;
```
Comment (M-Zuber) [inline MyHome.UI/Helpers/HelperMethods.cs]: If we have changed the amount type to be decimal, this should probably also change

File: MyHome.DataClasses/Expense.cs (hunk 1 above)
Comment (qchouleur) [inline MyHome.DataClasses/Expense.cs]: I will apply the second response in this post as the overhead mentioned in the comment is not a problem.

http://stackoverflow.com/questions/263400/what-is-the-best-algorithm-for-an-overridden-system-object-gethashcode

File: MyHome.DataRepositories/GeneralCategoryHandler.cs (hunk 2 above)
Comment (qchouleur) [inline MyHome.DataRepositories/GeneralCategoryHandler.cs]: I dislike using a string to retrieve a certain type of data, maybe the ui should be less "Dynamic" and we should use the type service directly.

File: MyHome.Services/CategoryService.cs (hunk 3 above)
Comment (qchouleur) [inline MyHome.Services/CategoryService.cs]: No need for that. Fixed.

File: MyHome.Services/MonthService.cs (hunk 5 above)
Comment (qchouleur) [inline MyHome.Services/MonthService.cs]: Meaning i needed to check and forgot it. it's fixed now.

File: MyHome.DataClasses/Expense.cs (hunk 1 above)
Comment (M-Zuber) [inline MyHome.DataClasses/Expense.cs]: :+1:

File: MyHome.DataRepositories/GeneralCategoryHandler.cs (hunk 2 above)
Comment (M-Zuber) [inline MyHome.DataRepositories/GeneralCategoryHandler.cs]: Does an `Enum` sound like a valid solution?

File: MyHome.Services/CategoryService.cs (hunk 3 above)
Comment (M-Zuber) [inline MyHome.Services/CategoryService.cs]: :+1:

File: MyHome.Services/MonthService.cs (hunk 5 above)
Comment (M-Zuber) [inline MyHome.Services/MonthService.cs]: :+1:

File: MyHome.Services/ExpenseCategoryService.cs (hunk 4 above)
Comment (M-Zuber) [inline MyHome.Services/ExpenseCategoryService.cs]: Never mind. I realized that the check is also made in the UI, so :+1:

**Comment by M-Zuber** [comment]
Looks good.
:shipit: 
Lets make it better!! :rocket:

**Pull request merged**